Files/View Models/FolderSettingsViewModel.cs

```diff
@@ -0,0 +1,359 @@
+
+        private class LayoutPreferences
+        {
+            public SortOption DirectorySortOption;
+            public SortDirection DirectorySortDirection;
+            public LayoutModes LayoutMode;
+            public int GridViewSize;
+
+            public static LayoutPreferences DefaultLayoutPreferences => new LayoutPreferences();
+
+            public LayoutPreferences()
+            {
+                this.LayoutMode = App.AppSettings.DefaultLayoutMode;
+                this.GridViewSize = App.AppSettings.DefaultGridViewSize;
+                this.DirectorySortOption = App.AppSettings.DefaultDirectorySortOption;
+                this.DirectorySortDirection = App.AppSettings.DefaultDirectorySortDirection;
+            }
+
+            public static LayoutPreferences FromCompositeValue(ApplicationDataCompositeValue compositeValue)
+            {
+                var layoutPreference = new LayoutPreferences();
+                layoutPreference.LayoutMode = (LayoutModes)(int)compositeValue[nameof(LayoutMode)];
+                layoutPreference.GridViewSize = (int)compositeValue[nameof(GridViewSize)];
+                layoutPreference.DirectorySortOption = (SortOption)(int)compositeValue[nameof(DirectorySortOption)];
+                layoutPreference.DirectorySortDirection = (SortDirection)(int)compositeValue[nameof(DirectorySortDirection)];
+                return layoutPreference;
+            }
+
+            public ApplicationDataCompositeValue ToCompositeValue()
+            {
+                var compositeValue = new ApplicationDataCompositeValue();
+                compositeValue[nameof(LayoutMode)] = (int)this.LayoutMode;
+                compositeValue[nameof(GridViewSize)] = this.GridViewSize;
+                compositeValue[nameof(DirectorySortOption)] = (int)this.DirectorySortOption;
+                compositeValue[nameof(DirectorySortDirection)] = (int)this.DirectorySortDirection;
+                return compositeValue;
+            }
+
+            public override bool Equals(object obj)
+            {
+                if (obj == null)
+                {
+                    return false;
+                }
+                if (obj == this)
+                {
+                    return true;
+                }
+                if (obj is LayoutPreferences prefs)
+                {
+                    return (
+                        prefs.LayoutMode == this.LayoutMode &&
+                        prefs.GridViewSize == this.GridViewSize &&
+                        prefs.DirectorySortOption == this.DirectorySortOption &&
+                        prefs.DirectorySortDirection == this.DirectorySortDirection);
+                }
+                return base.Equals(obj);
+            }
+
+            public override int GetHashCode()
+            {
+                return base.GetHashCode();
+            }
+        }
+
+        public enum LayoutModes
+        {
+            DETAILS_VIEW = 0,
```

Comment:
Can you align your code with our code style? By the way, each .cs file should contain only one class or enum.